dolphin/api/v1/storages.py

```diff
@@ -27,7 +51,53 @@ def show(self, req, id):
         return dict(name="Storage 2")
 
     def create(self, req, body):
-        return dict(name="Storage 3")
+        """
+        This function for registering the new storage device
+        :param req:
+        :param body: "It contains the all input parameters"
+        :return:
+        """
+        # Check if body is valid
+        if not self.is_valid_body(body, 'storages'):
+            msg = ("Storage entity not found in request body")
+            raise exc.HTTPUnprocessableEntity(explanation=msg)
+
+        storage = body['storages']
+
+        # validate the body has all required parameters
+        required_parameters = ('hostip', 'vendor', 'model', 'username', 'password')
+        validate_parameters(storage, required_parameters)
+
+        # validate the hostip
+        if not utils.is_valid_ip_address(storage['hostip'], ip_version='4'):
+            msg = ("Invalid hostip. Please provide a valid hostip")
+            LOG.error(msg)
+            raise exception.InvalidHost(msg)
+
+        # get dolphin.context. Later may be validated context parameters
+        context = req.environ.get('dolphin.context')
+
+        driver = drivermanager.DriverManager()
+        device_info = driver.register_storage(context, storage)
+        if device_info.get('status') == 'available':
```

Comment:
if get device failed, the driver manager should raise an exception.

Reply:
done